homeassistant/components/sensor/openuv.py

```diff
@@ -0,0 +1,113 @@
+
+ATTR_MAX_UV_TIME = 'time'
+
+
+async def async_setup_platform(
+        hass, config, async_add_devices, discovery_info=None):
+    """Set up the OpenUV binary sensor platform."""
+    if discovery_info is None:
+        return
+
+    openuv = hass.data[DOMAIN]
+
+    sensors = []
+    for sensor_type in discovery_info[CONF_MONITORED_CONDITIONS]:
+        name, icon, unit = SENSORS[sensor_type]
+        sensors.append(OpenUvSensor(openuv, sensor_type, name, icon, unit))
+
+    async_add_devices(sensors, True)
+
+
+class OpenUvSensor(OpenUvEntity):
+    """Define a binary sensor for OpenUV."""
+
+    def __init__(self, openuv, sensor_type, name, icon, unit):
+        """Initialize the sensor."""
+        super().__init__(openuv)
+
+        self._icon = icon
+        self._latitude = openuv.client.latitude
+        self._longitude = openuv.client.longitude
+        self._name = name
+        self._sensor_type = sensor_type
+        self._state = None
+        self._unit = unit
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def should_poll(self):
+        """Disable polling."""
+        return False
+
+    @property
+    def state(self):
+        """Return the status of the sensor."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return '{0}_{1}_{2}'.format(
+            self._latitude, self._longitude, self._sensor_type)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit
+
+    @callback
+    def _update_data(self):
+        """Update the state."""
+        self.async_schedule_update_ha_state(True)
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        async_dispatcher_connect(
+            self.hass, SENSOR_UPDATE_TOPIC, self._update_data)
+
+    async def async_update(self):
+        """Update the state."""
+        data = self.openuv.data[DATA_UV]['result']
+        if self._sensor_type == TYPE_CURRENT_OZONE_INDEX:
+            self._state = data['ozone']
+        elif self._sensor_type == TYPE_CURRENT_UV_INDEX:
+            self._state = data['uv']
+        elif self._sensor_type == TYPE_MAX_UV_INDEX:
+            self._state = data['uv_max']
+            self._attrs.update({
+                ATTR_MAX_UV_TIME: as_local(
+                    parse_datetime(data['uv_max_time']))
+            })
+        elif self._sensor_type in (TYPE_SAFE_EXPOSURE_TIME_1,
+                                   TYPE_SAFE_EXPOSURE_TIME_2,
+                                   TYPE_SAFE_EXPOSURE_TIME_3,
+                                   TYPE_SAFE_EXPOSURE_TIME_4,
+                                   TYPE_SAFE_EXPOSURE_TIME_5,
+                                   TYPE_SAFE_EXPOSURE_TIME_6):
+            kind = self._sensor_type[-1]
```

Comment:
Depending on the string value of a constant defined in home assistant for interaction with another API like this, could be considered error prone.

It would be safer to decouple the APIs using a dictionary.

Reply:
Got it. Let me give it a whirl and see if you are okay with the direction.

Reply:
I'd like to add a dict which maps home assistant sensor type constant to openuv kind.

With this dict we decouple the home assistant constants from the openuv api. Ie we can change the constant strings in home assistant without breaking the integration.

Reply:
Let me know if you're okay with what I propose.